doc/api/stream.md

```diff
@@ -2031,6 +2031,32 @@ added: REPLACEME
 * `streamWritable` {stream.Writable}
 * Returns: {WritableStream}
 
+### `stream.Duplex.from(src)`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `src` {{Stream|Blob|ArrayBuffer|string|Iterable|AsyncIterable|AsyncGeneratorFunction|AsyncFunction|Promise|WritableReadablePair}
+
+A utility method for creating duplex streams.
+
+* `Stream` converts writable stream into writable `Duplex` and readable stream
+  to `Duplex`.
+* `Blob` converts into readable `Duplex`.
+* `string` converts into readable `Duplex`.
+* `ArrayBuffer` converts into readable `Duplex`.
+* `AsyncIterable` converts into a readable `Duplex`. Cannot yield
+  `null`.
+* `AsyncGeneratorFunction` converts into a readable/writable transform `Duplex`.
+  Must take a source `AsyncIterable` as first parameter. Cannot yield
+  `null`.
+* `AsyncFunction` converts into a writable `Duplex`. Must return
+  either `null` or `undefined`
+* `WritableReadablePair ({ writable, readable })` converts `readable` and `writable` into `Stream` and
+  then combines them into `Duplex` where the `Duplex` will write to the `writable`
+  and read from the `readable`.
+* `Promise` converts into readable `Duplex`. Value `null` is ignored.
```

Comment:
Not blocking on this but I think this needs an example and motivation in the docs.